aws/resource_aws_s3_bucket.go

```diff
@@ -76,6 +76,14 @@ func resourceAwsS3Bucket() *schema.Resource {
 				Default:       "private",
 				Optional:      true,
 				ConflictsWith: []string{"grant"},
+				ValidateFunc: validation.StringInSlice([]string{
+					s3.BucketCannedACLPrivate,
+					s3.BucketCannedACLPublicRead,
+					s3.BucketCannedACLPublicReadWrite,
+					s3.BucketCannedACLAWSExecRead,
```

Comment:
I don't think this exists for buckets... For objects, there's `ObjectCannedACLAwsExecRead`.

Reply:
https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html#canned-acl

![image](https://user-images.githubusercontent.com/8305363/94368950-3c39f580-00a4-11eb-804b-756a4da69913.png)

It seems to say it works on both

Reply:
The SDK doesn't seem to support it. See https://docs.aws.amazon.com/sdk-for-go/api/service/s3/#pkg-constants. Or if you look at the code directly (https://raw.githubusercontent.com/aws/aws-sdk-go/master/service/s3/api.go), you can take a look at the `BucketCannedACL_Values()` function.

Reply:
I see you're correct. Those values are exist at the ObjectCannedACL_Values() function. I  dont know the behavior just yet, and would need to test, but in this case, would it be wise to add this validator in if the upstream SDK doesn't support all possible values for a bucket? Wouldn't want to return a validation error when the ACL is perfectly valid.

```
// ObjectCannedACL_Values returns all elements of the ObjectCannedACL enum
func ObjectCannedACL_Values() []string {
	return []string{
		ObjectCannedACLPrivate,
		ObjectCannedACLPublicRead,
		ObjectCannedACLPublicReadWrite,
		ObjectCannedACLAuthenticatedRead,
		ObjectCannedACLAwsExecRead,
		ObjectCannedACLBucketOwnerRead,
		ObjectCannedACLBucketOwnerFullControl,
	}
}
```